mlflow/pyfunc/__init__.py

```diff
@@ -984,6 +992,18 @@ def _predict_row_batch(predict_fn, args):
                     )
             pdf = pandas.DataFrame(data={names[i]: x for i, x in enumerate(args)}, columns=names)
 
+        # If the spark dataframe input column is array type,
+        # then in spark pandas_udf, the passed in arguments
+        # (`pd.dataframe` instance or `pd.Series`) contains numpy array values.
+        # Converting the numpy array values into list
+        # Because `PyFuncModel.predict` only accepts pandas dataframe
```

Comment:
This seems redundant. Can we make pyfunc predict work with numpy array values instead?

Reply:
My purpose is (1) in https://github.com/mlflow/mlflow/pull/7291#issuecomment-1324434216
but, final decision is up to you.

Reply:
I think this is different though. Can we not accept both list and numpy array in the cast and reshape function?

Reply:
Replied in https://github.com/mlflow/mlflow/pull/7291#issuecomment-1325876838